src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx

```diff
@@ -143,35 +170,65 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
-      setSnsManuallyDefinedFailureRate(undefined);
+      // SELECTED ESTIMATE => PREDICTED OR OPERATIONAL IS SELECTED
+      const iriOfSelectedValue = shapeToolData.selectedEstimate.iri;
+      const { predictionIri, operationalIri } = supertypes.reduce(
+        (acc, item) => {
+          if (item?.hasFailureRate?.prediction?.iri) acc.predictionIri = item.hasFailureRate.prediction.iri;
+          if (item?.hasFailureRate?.estimate?.iri) acc.operationalIri = item.hasFailureRate.estimate.iri;
+          return acc;
+        },
+        { predictionIri: "", operationalIri: "" },
+      );
+
+      if (iriOfSelectedValue === predictionIri) {
+        setSelectedRadioButton(RadioButtonType.Predicted);
+      } else if (iriOfSelectedValue === operationalIri) {
+        setSelectedRadioButton(RadioButtonType.Operational);
+      }
+      setSnsManuallyDefinedFailureRate(shapeToolData?.probability);
     } else {
       // NO SELECTED ESTIMATE => MANUAL IS SELECTED
       setSelectedRadioButton(RadioButtonType.Manual);
       if (shapeToolData?.probability) {
         setSnsManuallyDefinedFailureRate(shapeToolData.probability);
+        setExternalManuallyDefinedFailureRate(shapeToolData.probability);
       }
     }
-    let supertypes: any = shapeToolData?.supertypes?.supertypes;
-    if (supertypes) {
-      const objectWithHasFailureRate = supertypes.find((obj) => obj.hasOwnProperty("hasFailureRate"));
 
-      if (objectWithHasFailureRate?.hasFailureRate?.prediction?.value) {
-        setSnsPredictedFailureRate(objectWithHasFailureRate?.hasFailureRate?.prediction?.value);
-      }
-      if (objectWithHasFailureRate?.hasFailureRate?.estimate?.value) {
-        setSelectedRadioButton(RadioButtonType.Operational);
-        setSnsOperationalFailureRate(objectWithHasFailureRate?.hasFailureRate?.estimate?.value);
+    if (supertypes) {
+      for (let i = 0; i < supertypes.length; i++) {
+        const item = supertypes[i];
+        if (item?.hasFailureRate?.estimate?.value) {
+          setSnsOperationalFailureRate(item?.hasFailureRate?.estimate?.value);
+          setSnsOperationalIri(item?.hasFailureRate?.estimate?.iri);
+        }
+        if (item?.hasFailureRate?.prediction?.value) {
+          setSnsPredictedFailureRate(item?.hasFailureRate?.prediction?.value);
+          setSnsPredictedIri(item?.hasFailureRate?.prediction?.iri);
+        }
       }
     }
   }, [shapeToolData]);
 
   const basedFailureRate = shapeToolData?.supertypes?.supertypes?.hasFailureRate?.estimate?.value;
   const requiredFailureRate = shapeToolData?.supertypes?.hasFailureRate?.requirement?.upperBound;
 
+  const FailureRateBox = ({ value, label, rate, selected, classes }) => (
+    <Box display="flex" flexDirection="row" alignItems="center">
+      <FormControlLabel
+        value={value}
+        control={<Radio className={classes.black} />}
+        label={`${label}:`}
+        className={selected ? classes.black : classes.grey}
```

Comment:
We need to have semantic classes according to https://github.com/kbss-cvut/fta-fmea-ui/pull/367 and not colors directly injected here

Reply:
Renamed to selected and notSelected

Reply:
where did you rename it? You do not have to necessarily do separate issue here but put here the PR where you solved it

Reply:
When possible, I would try to use [MUI default styling variants](https://mui.com/material-ui/api/typography/) using the `variant` prop.

Reply:
Was replaced with classes: "selected", "notSelected", "editableValue", "notEditableValue"